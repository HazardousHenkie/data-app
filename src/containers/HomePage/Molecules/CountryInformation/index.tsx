import React from 'react'

import { createSelector } from 'reselect'
import { Reducer } from 'redux'
import { useSelector } from 'react-redux'

import Avatar from '@material-ui/core/Avatar'
import InfoIcon from '@material-ui/icons/Info'
import Card from '@material-ui/core/Card'
import CardHeader from '@material-ui/core/CardHeader'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import reducer from 'containers/HomePage/Molecules/CountryListItem/reducer'
import { useInjectReducer } from 'utils/redux-injectors'

import makeSelectCountry from 'containers/HomePage/Molecules/CountryListItem/selectors'

import { useTranslation } from 'react-i18next'

const stateSelector = createSelector(makeSelectCountry(), country => ({
    country
}))

const CountryInformation: React.FC = () => {
    const { t, i18n } = useTranslation('homePage')
    const { country } = useSelector(stateSelector)

    useInjectReducer({ key: 'country', reducer: reducer as Reducer })

    return (
        <Card data-testid="CountryInformationCard" variant="outlined">
            <CardHeader
                data-testid="CountryInformationCardHeader"
                avatar={
                    <Avatar
                        aria-label={t(
                            'homePage:countryInformation.countryInformation',
                            'Country Information'
                        )}
                    >
                        <InfoIcon />
                    </Avatar>
                }
                title={`${t(
                    'homePage:countryInformation.informationTitle',
                    'Information for'
                )}
                ${
                    i18n.language === 'en'
                        ? country.name
                        : country.translations[i18n.language]
                }`}
            />
            <CardContent data-testid="CountryInformationCardContent">
                {country.nativeName && (
                    <Typography
                        data-testid="countryNativeName"
                        variant="body1"
                        component="p"
                    >
                        <strong>
                            {t(
                                'homePage:countryInformation.nativeName',
                                'Native name:'
                            )}
                        </strong>
                        {` ${country.nativeName}`}
                    </Typography>
                )}

                {country.region && (
                    <Typography
                        data-testid="countryRegionName"
                        variant="body1"
                        component="p"
                    >
                        <strong>
                            {t('homePage:countryInformation.region', 'Region:')}
                        </strong>
                        {` ${country.region}`}
                    </Typography>
                )}

                {country.subregion && (
                    <Typography
                        data-testid="countrySubRegionName"
                        variant="body1"
                        component="p"
                    >
                        <strong>
                            {t(
                                'homePage:countryInformation.subregion',
                                'Subregion:'
                            )}
                        </strong>
                        {` ${country.subregion}`}
                    </Typography>
                )}

                {country.population && (
                    <Typography
                        data-testid="countryPopulation"
                        variant="body1"
                        component="p"
                    >
                        <strong>
                            {t(
                                'homePage:countryInformation.population',
                                'Population:'
                            )}
                        </strong>
                        {` ${country.population}`}
                    </Typography>
                )}

                {country.currencies &&
                    country.currencies[0] &&
                    country.currencies[0].code && (
                        <Typography
                            data-testid="countryCurrencies"
                            variant="body1"
                            component="p"
                        >
                            <strong>
                                {t(
                                    'homePage:countryInformation.currencies',
                                    'Currencies:'
                                )}
                            </strong>
                            {` ${country.currencies.map(
                                currency =>
                                    ` ${currency.code}(${currency.symbol})`
                            )}`}
                        </Typography>
                    )}
            </CardContent>
        </Card>
    )
}

export default CountryInformation
