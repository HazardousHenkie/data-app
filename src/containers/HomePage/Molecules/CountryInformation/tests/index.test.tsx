import React from 'react'
import renderer from 'react-test-renderer'

import MockingAppComponent from 'utils/testing/MockingAppComponent'
import CountryInformation from '../index'

describe('<FavoriteCountryButton />', () => {
    it('should render like snapshot', () => {
        const component = renderer.create(
            <MockingAppComponent>
                <CountryInformation />
            </MockingAppComponent>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
