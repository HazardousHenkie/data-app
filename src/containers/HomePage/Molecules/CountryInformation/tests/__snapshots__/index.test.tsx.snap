// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteCountryButton /> should render like snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="homePage:countryAdvisor.countryInformation"
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        homePage:countryAdvisor.advisoryTitle
                undefined
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    
    
    
    
    0
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <strong>
        homePage:countryInformation.languages
        :
      </strong>
        undefined(undefined)
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <strong>
        homePage:countryInformation.currencies
        :
      </strong>
        undefined(undefined)
    </p>
  </div>
</div>
`;
